### PR TITLE
Add gce-serial and gke-serial jobs

### PR DESCRIPTION
**Pull request by ikehz**
To start deflaking serial, disruptive, and restart tests.

Need to do this per #19681; looks like Reboot tests don't play with other tests.

### CODE REVIEW

**Comment by ihmccreery** [comment]
@k8s-bot test this again please

File: hack/jenkins/job-configs/kubernetes-e2e.yaml [hunk 1]
```diff
@@ -189,3 +189,29 @@
...
+    suffix:
+        - 'gce-serial':
+            description: 'Run [Serial], [Disruptive], and [Feature:Restart] tests on GCE using the latest successful build.'
+            timeout: 300
+            emails: '$DEFAULT_RECIPIENTS, ihmccreery@google.com'
+            test-owner: 'ihmccreery'
+    jobs:
+        - 'kubernetes-e2e-{suffix}'
+
+- project:
```
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-e2e.yaml]: Why do you need two projects? It looks like you can collapse these into one.

Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-e2e.yaml]: Done.

**Comment by ihmccreery** [comment]
@spxtr PTAL.

**Comment by spxtr** [comment]
LGTM. You can set the variables in the project, btw, so you don't need to repeat things like timeout. Not a big deal, though.

**Comment by ihmccreery** [comment]
Merging by hand to start getting results.  cc @k8s-oncall

File: hack/jenkins/e2e.sh [hunk 2]
```diff
@@ -507,6 +507,38 @@ case ${JOB_NAME} in
...
+          \[Feature:Restart\] \
+          ) --ginkgo.skip=$(join_regex_no_empty \
+	  \[Flaky\]
+          )"}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="e2e-serial"}
+    : ${PROJECT:="kubernetes-jkns-e2e-gce-serial"}
+    ;;
+
+  # Run the [Serial], [Disruptive], and [Feature:Restart] tests on GKE.
+  kubernetes-e2e-gke-serial)
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: we probably should be more consistent in our GKE job names. Some of the ones running from `master` are `gke-ci`, others are `gke`...

Comment (spxtr) [inline hack/jenkins/e2e.sh]: Agreed. I think the manner done here is consistent with the GCE jobs, no suffix means it's running from master.

**Pull request merged**